### PR TITLE
Add type hints that pass mypy checks

### LINKED ISSUE #1487: Wrong type hinting in ert-shared
**Context**
During the work in https://github.com/equinor/ert/pull/1452 I had to disable all type hinting errors for `ert-shared` because apparently there are some errors according to `mypy`.

**Steps to reproduce**

Remove the following lines from both `.mypy.ini` and `.mypy-strict.ini`:

```
[mypy-ert_shared.*]
ignore_errors = True
```

and run

`mypy ert3`.

**Desired outcome**
Drop the ignores as described above. Have the CI pass by making the necessary changes in `ert-shared`.

### PR DESCRIPTION
**Pull request by verveerpj**
**Issue**
Resolves #1487

### CODE REVIEW

File: ert_shared/ensemble_evaluator/config.py [hunk 1]
```diff
@@ -28,16 +28,17 @@ def find_open_port(lower=51820, upper=51840):
...
+        self.host: str = _get_ip_address()
+        self.port: int = find_open_port() if port is None else port
+        self.socket: socket.socket = bind_socket(self.host, self.port)
+        self.url: str = f"ws://{self.host}:{self.port}"
+        self.client_uri: str = f"{self.url}/client"
+        self.dispatch_uri: str = f"{self.url}/dispatch"
+
+    def get_socket(self) -> socket.socket:
+        # The use of _closed seems questionable.
+        if self.socket._closed:  # type: ignore
```
Comment (sondreso) [inline ert_shared/ensemble_evaluator/config.py:41]: Could you create an issue instead and refer to that? 🙂

**Review by sondreso — DISMISSED** [review]
LGTM, except for the small comment!

**Review by sondreso — APPROVED** [review]
LGTM!

**Comment by verveerpj** [comment]
Jenkins, test this please

**Pull request merged**